.build-tools/pkg/metadataschema/validators.go

```diff
@@ -135,6 +143,52 @@ func (c *ComponentMetadata) AppendBuiltin() error {
 				},
 			},
 		)
+	case mdutils.BindingType:
+		if c.Binding != nil {
+			if c.Metadata == nil {
+				c.Metadata = []Metadata{}
+			}
+
+			var direction string
+
+			switch {
+			case c.Binding.Input && c.Binding.Output:
+				direction = bindingDirectionBoth
+			case c.Binding.Input:
+				direction = bindingDirectionInput
+			case c.Binding.Output:
+				direction = bindingDirectionOutput
+			}
+
+			c.Metadata = append(c.Metadata,
+				Metadata{
+					Name:        bindingDirectionMetadataKey,
```

Comment:
Do you think we should include this option only if the binding is both input and output? Otherwise it can be confusing.

Reply:
This property was specifically designed to let users specify that a binding is not both input and output

Reply:
The metadata property "direction" should only be included if the component offers input binding capabilities. A few lines higher you use the `c.Binding.Input` and `c.Binding.Output` properties that indicate if the component is an input and/or output binding.

- If the component is both input and output (`c.Binding.Input && c.Binding.Output`):
   - Include the "direction" metadata property with allowed values (as **enum** type): `input`,`output`,`input,output` (no space)
- If the component is **just** input (`c.Binding.Input`):
   - Include the "direction" metadata property with only 1 allowed value (as **enum** type): `input` (adding this will make Dapr not perform the initial OPTIONS request)
- If the component is **just** output (`c.Binding.Output`):
   - Metadata property "direction" is not present